From db6bc96f762aa4c783ec0d06c150daab83ae9277 Mon Sep 17 00:00:00 2001
From: Aaron Leventhal <aleventhal@google.com>
Date: Wed, 24 Jan 2024 21:12:39 +0000
Subject: [PATCH] M121: [A11y] Remove PreSerializationConsistencyCheck in 6167 branch

This is a bug that appears to be fixed in M122. For M121, just comment
out the checks.

Fixed: 1520355
Change-Id: Ief1c13641bb4955bc6a4d83a03ae3f8d66520fa8
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5225301
Reviewed-by: Mark Schillaci <mschillaci@google.com>
Cr-Commit-Position: refs/branch-heads/6167@{#1632}
Cr-Branched-From: 222e786949e76e342d325ea0d008b4b6273f3a89-refs/heads/main@{#1233107}
---

diff --git a/third_party/blink/renderer/modules/accessibility/ax_object.cc b/third_party/blink/renderer/modules/accessibility/ax_object.cc
index a7b997f..68020cb 100644
--- a/third_party/blink/renderer/modules/accessibility/ax_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_object.cc
@@ -1342,8 +1342,6 @@
   node_data->role = ComputeFinalRoleForSerialization();
   node_data->id = AXObjectID();
 
-  PreSerializationConsistencyCheck();
-
   // Serialize a few things that we need even for ignored nodes.
   bool is_focusable = CanSetFocusAttribute();
   if (is_focusable)
@@ -7526,23 +7524,6 @@
   return common_ancestor;
 }
 
-// Extra checks that only occur during serialization.
-void AXObject::PreSerializationConsistencyCheck() {
-  CHECK(!IsDetached()) << "Do not serialize detached nodes: "
-                       << ToString(true, true);
-  // TODO(https://crbug.com/1480627): convert to CHECKs.
-  CHECK(AXObjectCache().IsFrozen());
-  CHECK(!NeedsToUpdateCachedValues());
-  CHECK(AccessibilityIsIncludedInTree())
-      << "Do not serialize unincluded nodes: " << ToString(true, true);
-#if defined(AX_FAIL_FAST_BUILD)
-  // A bit more expensive, so only check in builds used for testing.
-  CHECK_EQ(IsAriaHidden(), !!FindAncestorWithAriaHidden(this))
-      << "IsAriaHidden() doesn't match existence of an aria-hidden ancestor: "
-      << ToString(true);
-#endif
-}
-
 String AXObject::ToString(bool verbose, bool cached_values_only) const {
   // Build a friendly name for debugging the object.
   // If verbose, build a longer name name in the form of:
diff --git a/third_party/blink/renderer/modules/accessibility/ax_object.h b/third_party/blink/renderer/modules/accessibility/ax_object.h
index a6fd19f..b5b67e4 100644
--- a/third_party/blink/renderer/modules/accessibility/ax_object.h
+++ b/third_party/blink/renderer/modules/accessibility/ax_object.h
@@ -1379,9 +1379,6 @@
   const AtomicString& GetRoleAttributeStringForObjectAttribute(
       ui::AXNodeData* node_data);
 
-  // Extra checks that occur right before a node is evaluated for serialization.
-  void PreSerializationConsistencyCheck();
-
   // Returns a string representation of this object.
   // |cached_values_only| avoids recomputing cached values, and thus can be
   // used during UpdateCachedValuesIfNecessary() without causing recursion.
